File: core/src/main/java/org/tgoe/core/member/enums/MemberDosbGender.java
package org.tgoe.core.member.enums;

public enum MemberDosbGender {
	MALE("m"),
	FEMALE("w"),
	DIVERSE("d");
	
	private String key;
	
	private MemberDosbGender(String key) {
		this.key=key;
	}
	
	public String getKey() {
		return key;
	}
	
	/**
	 * Find enum value matching the key. Uses case-sensitive string comparison.
	 * 
	 * @param key Key to search for.
	 * @return Enumeration value or NULL in case key does not exist.
	 */
	public static MemberDosbGender findByKey(String key) {
		for( MemberDosbGender p : MemberDosbGender.values() ) {
			if( p.getKey().equals(key) ) return p;
		}
		
		return null;
	}	
}

File: src/TgoeSrv/Member/Enums/MemberDosbGender.php
<?php
namespace TgoeSrv\Member\Enums;

enum MemberDosbGender : string
{
    case MALE = 'm';
    case FEMALE = 'w';
    case DIVERSE = 'd';
    case UNKNOWN = 'u';
    
    /**
     * Find enum value matching the key. Uses ignore case string comparison.
     *
     * @param string $key Key to search for.
     * @return DosbSport Enumeration value or UNKNOWN in case key does not exist.
     */
    public static function findByKey(string $key): ?MemberDosbGender
    {
        foreach( MemberDosbGender::cases() as $e) {
            if( $e->value == $key ) return $e;
        }
        
        return self::UNKNOWN;
    }
}

